packages/bs-neo-legacy/src/services/BSNeoLegacy.ts

```diff
@@ -259,66 +271,27 @@ export class BSNeoLegacy<BSName extends string = string>
-
-    if (gasAmountNumber) {
-      // Two transfers fee and one transfer fee left over
-      const allNep17TransfersFee = BSNeoLegacyConstants.MIGRATION_NEO3_NEP_17_TRANSFER_FEE * 3
+  async migrate({ account, neo3Address, neoLegacyMigrationAmounts }: MigrateParams<BSName>): Promise<string> {
+    if (!neo3Address) throw new Error('Must have address')
 
-      // Necessary to calculate the COZ fee
-      const gasAmountNumberLessAllNep17TransfersFee = gasAmountNumber - allNep17TransfersFee
-
-      // Example: ~0.06635710 * 0.01 = ~0.00066357
-      const cozFee = gasAmountNumberLessAllNep17TransfersFee * BSNeoLegacyConstants.MIGRATION_NEO3_COZ_FEE
-
-      // Example: ~0.06635710 - ~0.00066357 = ~0.06569352
-      const gasAmountNumberLessCozFee = gasAmountNumberLessAllNep17TransfersFee - cozFee
-
-      const allGasFeeNumberThatUserWillPay = cozFee + BSNeoLegacyConstants.MIGRATION_NEO3_NEP_17_TRANSFER_FEE * 2
-      const allGasAmountNumberThatUserWillReceive =
-        gasAmountNumberLessCozFee + BSNeoLegacyConstants.MIGRATION_NEO3_NEP_17_TRANSFER_FEE
-
-      response.gasMigrationTotalFees = formatNumber(allGasFeeNumberThatUserWillPay, this.GAS_ASSET.decimals)
-      response.gasMigrationAmount = formatNumber(allGasAmountNumberThatUserWillReceive, this.GAS_ASSET.decimals)
-    }
-
-    if (neoAmountNumber) {
-      response.neoMigrationTotalFees = formatNumber(
-        Math.ceil(neoAmountNumber * BSNeoLegacyConstants.MIGRATION_NEO3_COZ_FEE),
-        this.NEO_ASSET.decimals
-      )
-      response.neoMigrationAmount = formatNumber(
-        neoAmountNumber - Number(response.neoMigrationTotalFees),
-        this.NEO_ASSET.decimals
-      )
-    }
-
-    return response
-  }
-
-  async migrateToNeo3({ account, address }: MigrateToNeo3Params<BSName>): Promise<string> {
-    if (!address) throw new Error('Must have address')
-
-    const { gasAmountNumber, neoAmountNumber } = await this.getMigrationNeo3ValidatedData({ account })
     const { neonJsAccount, signingCallback } = await this.#generateSigningCallback(account)
     const provider = new api.neoCli.instance(this.network.url)
     const intents: ReturnType<typeof api.makeIntent> = []
 
-    if (gasAmountNumber)
+    if (neoLegacyMigrationAmounts.gasBalance)
       intents.push(
-        ...api.makeIntent({ [this.GAS_ASSET.symbol]: gasAmountNumber }, BSNeoLegacyConstants.MIGRATION_NEO3_COZ_ADDRESS)
+        ...api.makeIntent(
+          { [this.GAS_ASSET.symbol]: Number(neoLegacyMigrationAmounts.gasBalance.amount) },
+          BSNeoLegacyConstants.MIGRATION_NEO3_COZ_ADDRESS
+        )
       )
 
-    if (neoAmountNumber)
+    if (neoLegacyMigrationAmounts.neoBalance)
       intents.push(
-        ...api.makeIntent({ [this.NEO_ASSET.symbol]: neoAmountNumber }, BSNeoLegacyConstants.MIGRATION_NEO3_COZ_ADDRESS)
+        ...api.makeIntent(
+          { [this.NEO_ASSET.symbol]: Number(neoLegacyMigrationAmounts.neoBalance.amount) },
```

Comment:
Is there `neoLegacyMigrationAmounts.neoBalance.amountNumber`?

Reply:
No. I am using the BalanceResponse interface